Makefile

```diff
@@ -596,6 +630,20 @@ startup_files: include_dirs $(LIBC_BOTTOM_HALF_CRT_OBJS)
 	mkdir -p "$(SYSROOT_LIB)" && \
 	cp $(LIBC_BOTTOM_HALF_CRT_OBJS) "$(SYSROOT_LIB)"
 
+# TODO: As of this writing, wasi_thread_start.s uses non-position-independent
+# code, and I'm not sure how to make it position-independent.  Once we've done
+# that, we can enable libc.so for the wasi-threads build.
+ifneq ($(THREAD_MODEL), posix)
+LIBC_SO = \
+	$(SYSROOT_LIB)/libc.so \
+	$(SYSROOT_LIB)/libwasi-emulated-mman.so \
+	$(SYSROOT_LIB)/libwasi-emulated-process-clocks.so \
+	$(SYSROOT_LIB)/libwasi-emulated-getpid.so \
+	$(SYSROOT_LIB)/libwasi-emulated-signal.so
```

Comment:
I'm not sure having 5 separate shared libraries here is the right way to go.

Could we maybe start by simply not included these optional parts in the shared library build?   For now they could continue to be linked in statically if needed, and we can maybe come up with a more advanced solution later, as needed?

Reply:
We'd still need to build separate libraries using `-fPIC` (e.g. `libwasi-emulated-mman-pic.a`).  Or I guess if folks are okay with using `-fPIC` unconditionally for the relevant .o files, we could do that.

Reply:
I was thinking the they would be linked into the main binary and wouldn't need `-fPIC`.   But that would limit their usefulness I guess.   This approach just seems a little overly complex, which is why I suggested punting on this for a while.  

An alternative approach, for example, would be to have a single libc-emulated.so or something like that.

Reply:
FWIW, libraries sometimes need these features, too, so I wouldn't want to restrict them to the main binary.

I'd be in favor of a single libc-emulated.so.  @yamt would that be okay with you?

Reply:
I went ahead and combined the libraries into a single `libwasi-emulated.so`.

Reply:
> I'd be in favor of a single libc-emulated.so. @yamt would that be okay with you?

i'm not sure if it's a good idea.
it would make try-link style configure scripts detect all of the emulated stuff
once you link the single libc-emulated.so.
also, what to do about _WASI_EMULATED_xxx macros?
i feel it's conservative to mirror what we do for static libs at least for the first step.

Reply:
To sum up the conversation so far:

- Initially, I bundled everything into libc.so for simplicity
- @yamt prefers we follow the pattern used for the static libraries and build a separate libwasi-emulated-*.so corresponding to each libwasi-emulated-*.a file
- @sbc100 is concerned that building so many .so files is overly complex and would prefer a single libwasi-emulated.so

Personally, I just want to get this PR merged, so I'm fine with any of the above.  If the maintainers of this project can come to a decision on this, I'm happy to implement any changes required.

Reply:
I don't feel strongly about the multiple `.so` files and in fact I can see value in mirroring the `.a` names.   Sorry for the noise.  

I wish we didn't need all those special `.so` files but I guess since we decided to make the emulation optional we don't have much choice.

Reply:
Thanks, @sbc100.  I've reverted the change the combined the `libwasi-emulated-*.so` into a single library, so we're back to four separate ones.